lib/src/collections.dart

```diff
@@ -24,3 +25,26 @@ E? firstOrNull<E>(Iterable<E> coll) {
   var it = coll.iterator;
   return it.moveNext() ? it.current : null;
 }
+
+/// Returns the value at [path] in a nested collection.
+/// Takes a [path] `List` and a collection [coll].
+/// Traverses `Map` (by key), `List` (by index), and `Iterable` (by index) values.
+dynamic getIn(List<dynamic> path, dynamic coll) {
```

Comment:
Technically this should probably be a static method in a class named CollectionsUtil:
https://github.com/jesims/docs/blob/master/Coding-Conventions/Dart.md#utilities

Although given this would be a breaking change to all the other functions, I am happy if we just want to live with it not matching conventions, given that they seem to be pretty high level, commonly used utilities which might benefit from not having to type as much.

Reply:
Yeah, not keen for _more_ breaking changes. Moved new `getIn` to a Util class. Added comments for other files